lib/mongodb-filter-wrapper.js

```diff
@@ -0,0 +1,155 @@
+'use strict';
+
+class MongoDBFilterWrapper {
+	static get filtersAllowed() {
+		return {
+			equal: '$eq',
+			not: '$ne',
+			greater: '$gt',
+			greaterOrEqual: '$gte',
+			lesser: '$lt',
+			lesserOrEqual: '$lte',
+			in: '$in',
+			notIn: '$nin',
+			reg: '$regex',
+			all: '$all'
+		};
+	}
+
+	/**
+		* Enter door to filter process
+		* @param model {Object} Model that will use the filter
+		* @param filters {Object} Filters defined as parameters
+		* @returns {Object} with the filter parsed
+		*/
+	static process(model, filters) {
+
+		// Validate filters given
+		if(typeof filters !== 'object' || (Array.isArray(filters) && filters.length === 0) || Object.entries(filters).length === 0)
+			return {};
+
+		// Get model filters
+		const modelFields = model.constructor.fields || {};
+
+		return this.getParsedFilters(filters, modelFields);
+	}
+
+	/**
+		* Validates filters given and loop calling to parse
+		* @param filters {Object} Filters defined as parameters
+		* @param modelFields {Object} Model that will use the filter
+		* @returns {Array|Object} The filter parsed to use in MongoDB
+		*/
+	static getParsedFilters(filters, modelFields) {
+		const orCondition = Array.isArray(filters);
+
+		let parsedFilters = orCondition ? [] : {};
+
+		// Only arrays as a filter
+		if(filters !== undefined && typeof filters === 'object' && Object.keys(filters).length > 0 && !Array.isArray(filters))
```

Comment:
```suggestion
		if(typeof filters === 'object' && !Array.isArray(filters) && Object.keys(filters).length)
```
No es necesario validar si `filters !== undefined` porque ya estas validando que el `typeof` sea `'object'`, por otro lado , antes de verificar si `filters` es un array estas consultando las keys del objeto y no es necesario verificar si el `Object.keys(filters).length` es mayor a 0 porque ya de por si da `false` si es menor o igual a 0